packages/rest/src/session/doc/ISession.ts

```diff
@@ -93,14 +93,14 @@ export interface ISession {
     base64EncodedAuth?: string;
 
     /**
-     * e.g. LTPA2
+     * Type of token in `tokenValue`, e.g. LTPA2
      * @type {string}
      * @memberof ISession
      */
-    tokenType?: SessConstants.TOKEN_TYPE_CHOICES;
+    tokenType?: string;
```

Comment:
Is it possible to define the property as:

    tokenType?: SessConstants.TOKEN_TYPE_CHOICE | string;

I know that this would result in no better constraint than just string. However, it could provide documentation value. A programmer can find the set of values built into imperative, and use a symbolic constant for its value.

Another thought: is a value of "guid" generally valuable to others? Should we add "guid" to the list of SessConstants.TOKEN_TYPE_CHOICES?

If the answer to both of these questions is "no" or "impractical", then we should just commit the code the way it is.

Reply:
If we implement the first suggestion, it won't make a difference in JSDoc:
![image](https://user-images.githubusercontent.com/50887074/85610746-45d73b80-b625-11ea-8af8-8592559f824b.png)

Reply:
I like the idea. It would probably be very helpful for developers. As for guid, I am not familiar enough to make a decision there.

Reply:
I meant that a programmer trying to program an appropriate value would see SessConstants.TOKEN_TYPE_CHOICES, look up its definition in SessConstants.ts, and find:

TOKEN_TYPE_LTPA = "LtpaToken2"
TOKEN_TYPE_JWT = "jwtToken"
TOKEN_TYPE_APIML = "apimlAuthenticationToken"

I would then write code using TOKEN_TYPE_APIML.

Reply:
That is also true. I don't see a problem with having both in there, and it could reduce problems developers have down the line if they decide to look into what the TOKEN_TYPE_CHOICES values are.

Reply:
> adding "custom-tokenType" values

I don't see this a a sustainable practice (i.e. adding every single custom `tokenType` value to imperative) just to allow for a functionality that used to be there before.

> Having `TOKEN_TYPE_CHOICES | string`

Since it does not affect the jsdoc, I believe it won't affect the generated typedoc documentation.